packages/php/email-editor/src/Integrations/Core/Renderer/Blocks/class-audio.php

```diff
@@ -0,0 +1,193 @@
+		if ( empty( $url ) ) {
+			return '';
+		}
+
+		// Validate URL type and format.
+		if ( ! str_starts_with( $url, 'data:audio/' ) &&
+			! str_starts_with( $url, '/' ) &&
+			! str_starts_with( $url, 'https://' ) ) {
+			// Reject everything else (http://, ftp://, relative paths, etc.).
+			return '';
+		}
+
+		// For HTTPS URLs, validate with wp_http_validate_url.
+		if ( str_starts_with( $url, 'https://' ) && ! wp_http_validate_url( $url ) ) {
+			return '';
+		}
+
+		// Get spacing from email_attrs for better consistency with core blocks.
+		$email_attrs        = $parsed_block['email_attrs'] ?? array();
+		$table_margin_style = '';
+
+		if ( ! empty( $email_attrs ) && class_exists( '\WP_Style_Engine' ) ) {
+			// Get margin for table styling.
+			$table_margin_style = \WP_Style_Engine::compile_css( array_intersect_key( $email_attrs, array_flip( array( 'margin' ) ) ), '' );
+		}
+
+		$icon_image = $this->get_audio_icon_url();
+		$label      = ! empty( $attr['label'] ) ? $attr['label'] : __( 'Listen to the audio', 'woocommerce' );
+
+		// Add duration to label if available.
+		if ( ! empty( $length ) ) {
+			$label .= ' (' . esc_html( (string) $length ) . ')';
+		}
+
+		$audio_url = esc_url( $url );
+
+		// Define pill-style colors and styling.
+		$background_color = '#f6f7f7';
+		$border_color     = '#AAA';
+		$icon_size        = '18px';
+		$font_size        = '14px';
+
+		// Generate the icon content.
+		$icon_content = sprintf(
+			'<a href="%1$s" rel="noopener nofollow" target="_blank" style="padding: 0.25em; padding-left: 17px; display: inline-block; vertical-align: middle;"><img height="%2$s" src="%3$s" style="display:block;margin-right:0;vertical-align:middle;" width="%2$s" alt="%4$s"></a>',
+			$audio_url,
+			esc_attr( $icon_size ),
+			esc_url( $icon_image ),
+			// translators: %s is the audio player icon.
+			sprintf( __( '%s icon', 'woocommerce' ), __( 'Audio', 'woocommerce' ) )
+		);
+		$icon_content = Table_Wrapper_Helper::render_table_cell( $icon_content, array( 'style' => sprintf( 'vertical-align:middle;font-size:%s;', $font_size ) ) );
+
+		// Generate the label content.
+		$label_content    = sprintf(
+			'<a href="%1$s" rel="noopener nofollow" target="_blank" style="text-decoration:none; padding: 0.25em; padding-right: 17px; display: inline-block;"><span style="margin-left:.5em;margin-right:.5em;font-weight:bold"> %2$s </span></a>',
+			$audio_url,
+			esc_html( $label )
+		);
+		$label_cell_style = sprintf(
+			'vertical-align:middle;font-size:%s;',
+			$font_size
+		);
+		$label_content    = Table_Wrapper_Helper::render_table_cell( $label_content, array( 'style' => $label_cell_style ) );
+
+		// Combine icon and label tables.
+		$audio_content = $icon_content . $label_content;
+
+		// Create the main pill-style table.
+		$main_table_styles = sprintf(
+			'background-color: %s; border-radius: 9999px; display: inline-table; float: none; border: 1px solid %s; border-collapse: separate;',
```

Comment:
Nit: The `display: inline-table;` seems to be causing some extra space below the table. Is it really needed?

<img width="590" height="103" alt="Hu1uHHpu1shHrl8C" src="https://github.com/user-attachments/assets/c43340b0-27e5-4bbd-b4ae-ed55ca3c816b" />

Reply:
Doesn't look like it! Removed in https://github.com/woocommerce/woocommerce/pull/60813/commits/6f0b9dbba17d11611ded9ecd23059061d899d18a